odoo/fields.py

```diff
@@ -978,6 +978,9 @@ def determine_value(self, record):
                 self.compute_value(record)
             else:
                 recs = record._in_cache_without(self)
+                if len(recs) > PREFETCH_MAX:
+                    recs = recs[:PREFETCH_MAX] | record
+                recs = recs.with_prefetch(record._prefetch)
```

Comment:
@KangOl I believe this line is wrong.  I think that `recs` should create a new prefetching set, initially restricted to `recs` itself.  Using the prefetching set of `record` might force the prefetching of unwanted records.  What do you think?